gemrb/core/GUI/WindowManager.cpp

```diff
@@ -91,12 +92,14 @@ WindowManager::~WindowManager()
 
 void WindowManager::DestroyWindows(WindowList& list)
 {
-	WindowList::iterator it = list.begin();
-	for (; it != list.end(); ++it) {
-		Window* win = *it;
+	while (!list.empty()) {
```

Comment:
interesting, still don't understand why, but I'll merge this. I do think doing the `clear()` in `DestroyWindows` would be better, but not a reason to hold back merging.